### DISCUSSION TITLE
`Vector2i` and `Rect2i` for `BitMap` and `Image`

### DISCUSSION THREAD

**Discussion by sent44** [godotengine/godot-proposals#4181] [Scripting]
Parameter and return type of functions of these classes should change to integer version as it is introduce in Godot 4. I don't see reason it is not to and it should be discourage to use real type.

Also where are `PackedVector2iArray` and `PackedVector3iArray`

**Comment by Mickeon** [comment]
> Also where are `PackedVector2iArray `and `PackedVector3iArray`

May be worth of its own proposal, but I'm not sure.
In 3.4.3, at least from what I've seen, cell data in a TileMap is all stored in a PoolIntArray and read as groups of 3, I believe... 
* cell's `x` & `y`, 
* tile id & flags, 
* autotile `x` and `y`.

For that reason I don't see the memory benefit, or much of a reason to add them other than they would *technically* make sense.

**Reply by sent44** [reply]
Narrowing conversion warning is annoying. In Godot 3.x I used a lot of `int(position.x)`. I usually used it like 2-4 times in a line, which make it hard to read the code on something I already knew it was int. (Or stack of `warning-ignore` before those codes) Not just only `make sense`.
Just add `i` after some `Position2` and `Size2` might solve this. `Position2i` and `Size2i` are already exist in Godot 4, I don't know where they are defined, but I saw their usage in a lot of classes when I searched.

**Reply by Mickeon** [reply]
I think this is a bit confused. `Position2i` and `Size2i` are not a data structure in Godot 4, and if they are, they're both simple `Vector2i`.

**Reply by sent44** [reply]
It is, like [`Image.set_pixelv()`](https://github.com/godotengine/godot/blob/894e2fddda029247e6e614f4a4e11d4b7fd8176d/core/io/image.cpp#L2972), in core internally that will translate to `Vector2i` in GDScript.
(set_pixelv and get_pixelv were only 2 functions in Image that changed to int version)
